### ISSUE TITLE
(1.12.2) Ice & Fire and NetherEx Support Not Working At All

### ISSUE THREAD

**Issue by TechnoMysterio** [OreCruncher/DynamicSurroundings#276]
I'm not sure why, but all of the config settings for Ice & Fire and NetherEx (even the ones present before the recent update) do not seem to be applying to any of the items or blocks, so armour and tools don't make the correct sounds, blocks don't have the correct footsteps etc.

MC: 1.12.2
Dynamic Surroundings: 1.12.2-3.4.9.7
Forge: 14.23.2.2624

**Comment by TechnoMysterio** [comment]
After a little testing, this also seems to apply to Actually Additions as well, with metal casing not producing metal footsteps, despite it being given that material in the config.

**Comment by OreCruncher** [comment]
Turn on DS debug tracing and startup the client.  See if there is an error listed in the log.

**Comment by TechnoMysterio** [comment]
Would that be 'Enable Debug Logging' in the config and 'latest.log' in the logs folder?

**Comment by OreCruncher** [comment]
Yep

**Comment by TechnoMysterio** [comment]
Not gonna lie, looking through these files is like wading through treacle, so I honestly can't tell you if there is an error or not. Is there anything in particular I could use ctrl+f to find?

**Comment by OreCruncher** [comment]
Search for "gson".  Essentially looking for parse errors.  As for Actually Additions it appears they changed the block names. :\

**Comment by TechnoMysterio** [comment]
Well there's some stuff related to the various json files in the dynamic surrounding jar (iceandfire.json, bloodmagic.json), but I don't know if there are any errors relating to them.

**Comment by TechnoMysterio** [comment]
Hmm, there is this in between some json files being processed:

`[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: com.google.gson.JsonSyntaxException: com.google.gson.stream.MalformedJsonException: Use JsonReader.setLenient(true) to accept malformed JSON at line 62 column 3 path $
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at com.google.gson.Gson.assertFullConsumption(Gson.java:863)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at com.google.gson.Gson.fromJson(Gson.java:826)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at org.blockartistry.lib.JsonUtils.load(JsonUtils.java:67)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at org.blockartistry.DynSurround.data.xface.DataScripts.loadFromStream(DataScripts.java:48)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at org.blockartistry.DynSurround.registry.RegistryManager.process(RegistryManager.java:135)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at org.blockartistry.DynSurround.registry.RegistryManager.lambda$null$4(RegistryManager.java:163)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at java.util.ArrayList.forEach(Unknown Source)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at org.blockartistry.DynSurround.registry.RegistryManager.lambda$reload$5(RegistryManager.java:161)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(Unknown Source)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(Unknown Source)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at java.util.stream.AbstractPipeline.copyInto(Unknown Source)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(Unknown Source)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(Unknown Source)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at java.util.stream.AbstractPipeline.evaluate(Unknown Source)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at java.util.stream.ReferencePipeline.forEach(Unknown Source)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at org.blockartistry.DynSurround.registry.RegistryManager.reload(RegistryManager.java:160)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at org.blockartistry.DynSurround.client.ClientRegistry.<clinit>(ClientRegistry.java:81)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at org.blockartistry.DynSurround.client.sound.SoundEngine.getVolumeScale(SoundEngine.java:431)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at org.blockartistry.DynSurround.client.sound.SoundEngine.getClampedVolume(SoundEngine.java:447)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at net.minecraft.client.audio.SoundManager.func_188770_e(SoundManager.java)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at net.minecraft.client.audio.SoundManager.func_148611_c(SoundManager.java:401)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at org.blockartistry.DynSurround.client.sound.SoundEngine.playSound(SoundEngine.java:258)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at org.blockartistry.DynSurround.client.gui.HumDinger.onGuiOpen(HumDinger.java:61)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_1202_HumDinger_onGuiOpen_GuiOpenEvent.invoke(.dynamic)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:179)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at net.minecraft.client.Minecraft.func_147108_a(Minecraft.java:982)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at net.minecraft.client.Minecraft.func_71384_a(Minecraft.java:544)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:377)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at net.minecraft.client.main.Main.main(SourceFile:123)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at java.lang.reflect.Method.invoke(Unknown Source)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at java.lang.reflect.Method.invoke(Unknown Source)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:196)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:231)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at org.multimc.EntryPoint.listen(EntryPoint.java:143)
[18:04:22] [main/INFO] [STDERR/]: [org.blockartistry.lib.logging.ModLog:error:109]: 	at org.multimc.EntryPoint.main(EntryPoint.java:34)
[18:04:22] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:-1]: Caused by: com.google.gson.stream.MalformedJsonException: Use JsonReader.setLenient(true) to accept malformed JSON at line 62 column 3 path $
[18:04:22] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:-1]: 	at com.google.gson.stream.JsonReader.syntaxError(JsonReader.java:1559)
[18:04:22] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:-1]: 	at com.google.gson.stream.JsonReader.checkLenient(JsonReader.java:1401)
[18:04:22] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:-1]: 	at com.google.gson.stream.JsonReader.doPeek(JsonReader.java:542)
[18:04:22] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:-1]: 	at com.google.gson.stream.JsonReader.peek(JsonReader.java:425)
[18:04:22] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:-1]: 	at com.google.gson.Gson.assertFullConsumption(Gson.java:859)
[18:04:22] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:-1]: 	... 45 more`

**Comment by OreCruncher** [comment]
Between which Jsons?  I&F and BM?

**Comment by TechnoMysterio** [comment]
It's preceded by:

`[18:58:17] [main/INFO] [dsurround/]: Unable to locate block for blockMap 'bloodmagic:BlockTeleposer'
[18:58:17] [main/INFO] [dsurround/]: Unable to locate block for blockMap 'bloodmagic:BlockWall1'
[18:58:17] [main/INFO] [dsurround/]: Unable to locate block for blockMap 'bloodmagic:BlockWall1'
[18:58:17] [main/INFO] [dsurround/]: Loading [dsurround:data/botania.json] <- [dsurround]
[18:58:17] [main/INFO] [dsurround/]: Loading [dsurround:data/ceramics.json] <- [dsurround]
[18:58:17] [main/INFO] [dsurround/]: Loading [dsurround:data/toughasnails.json] <- [dsurround]
[18:58:17] [main/INFO] [dsurround/]: Loading [dsurround:data/dynamictrees.json] <- [dsurround]
[18:58:17] [main/INFO] [dsurround/]: Loading [dsurround:data/enderstorage.json] <- [dsurround]
[18:58:17] [main/INFO] [dsurround/]: Unable to locate block for blockMap 'enderstorage:enderStorage'
[18:58:17] [main/ERROR] [blockartistry.lib/]: Unable to process Json from stream`

And followed by:

`[18:58:17] [main/INFO] [dsurround/]: Loading [dsurround:data/iceandfire.json] <- [dsurround]
[18:58:17] [main/INFO] [dsurround/]: Loading [dsurround:data/ironchest.json] <- [dsurround]
[18:58:17] [main/INFO] [dsurround/]: Loading [dsurround:data/randomthings.json] <- [dsurround]
[18:58:17] [main/INFO] [dsurround/]: Loading [dsurround:data/rustic.json] <- [dsurround]
[18:58:17] [main/INFO] [dsurround/]: Loading [dsurround:data/simplecorn.json] <- [dsurround]
[18:58:17] [main/INFO] [dsurround/]: Loading [dsurround:data/storagedrawers.json] <- [dsurround]
[18:58:17] [main/INFO] [dsurround/]: Loading [dsurround:data/streams.json] <- [dsurround]
[18:58:17] [main/INFO] [dsurround/]: Loading [dsurround:data/tcomplement.json] <- [dsurround]
[18:58:17] [main/INFO] [dsurround/]: Loading [dsurround:data/totemic.json] <- [dsurround]`

**Comment by TechnoMysterio** [comment]
Ah, that's interesting, when I search for 'netherex.json' it comes up with no results, so that one might not be being loaded.

**Comment by TechnoMysterio** [comment]
I think I've figured out why the Ice & Fire config file wasn't working: when I updated it to include item sounds, I wrote it wrong. I edited it, so hopefully it's fixed now. I've also edited the Actually Additions and Blood Magic files so that the block names are now correct for 1.12.2, and I altered the Tinkers' File so the crossbow sounds are fixed. Download below:

https://mega.nz/#!JGRw1bCS!fXqSbhkAbadd2eg1uhdFeyp279Sue9CDA6iiOD2ynhw

Now we'll just have to see what I've buggered up this time...
